modules/face/test/test_face_align.cpp

```diff
@@ -0,0 +1,117 @@
+using namespace cv;
+using namespace cv::face;
+
+TEST(CV_Face_FacemarkKazemi, can_create_default) {
+    Ptr<FacemarkKazemi> facemark;
+    string filename = cvtest::findDataFile("face/lbpcascade_frontalface_improved.xml", true);
+    CascadeClassifier face_cascade;
+    EXPECT_TRUE(face_cascade.load(filename));
+    EXPECT_NO_THROW(facemark = createFacemarkKazemi(face_cascade));
+    EXPECT_FALSE(facemark.empty());
+}
+
+TEST(CV_Face_FacemarkKazemi, can_loadTrainingData) {
+    string filename = cvtest::findDataFile("face/lbpcascade_frontalface_improved.xml", true);
+    CascadeClassifier face_cascade;
+    EXPECT_TRUE(face_cascade.load(filename));
+    Ptr<face::FacemarkKazemi> facemark;
+    EXPECT_NO_THROW(facemark = createFacemarkKazemi(face_cascade));
+    vector<String> filenames;
+    filename = cvtest::findDataFile("face/1.txt", true);
+    filenames.push_back(filename);
+    filename = cvtest::findDataFile("face/2.txt", true);
+    filenames.push_back(filename);
+    vector<String> imagenames;
+    vector< vector<Point2f> > trainlandmarks;
+    //Test getData function
+    EXPECT_NO_THROW(facemark->getData(filenames,trainlandmarks,imagenames));
+    EXPECT_TRUE(facemark->getData(filenames,trainlandmarks,imagenames));
+    vector<Mat> trainimages;
+    for(unsigned long i=0;i<imagenames.size();i++){
+        string img = cvtest::findDataFile(imagenames[i], true);
+        Mat src = imread(img);
+        EXPECT_TRUE(!src.empty());
+        trainimages.push_back(src);
+    }
+    //Test scaleData function
+    EXPECT_NO_THROW(facemark->scaleData(trainlandmarks,trainimages));
+    EXPECT_TRUE(facemark->scaleData(trainlandmarks,trainimages));
+    //Test calcMean function
+    EXPECT_NO_THROW(facemark->calcMeanShape(trainlandmarks,trainimages));
+    EXPECT_TRUE(facemark->calcMeanShape(trainlandmarks,trainimages));
+    //Test train function
+    string configfile_name = cvtest::findDataFile("face/config.xml", true);
+    EXPECT_NO_THROW(facemark->train(trainimages,trainlandmarks,configfile_name));
+    filename = cvtest::findDataFile("face/100032540_1.jpg", true);
+    Mat src = imread(filename);
+    vector<Rect> faces;
+    EXPECT_NO_THROW(facemark->getFaces(src,faces));
+    vector< vector<Point2f> > initialshapes;
+    EXPECT_NO_THROW(facemark->getMeanShapeRelative(faces,initialshapes));
+}
+TEST(CV_Face_FacemarkKazemi, can_detect_landmarks) {
+    Ptr<FacemarkKazemi> facemark;
+    string filename = cvtest::findDataFile("face/lbpcascade_frontalface_improved.xml", true);
+    string imgname = cvtest::findDataFile("face/detect.jpg");
+    string modelfilename = cvtest::findDataFile("face/face_landmarks_model.dat",true);
+    Mat img = imread(imgname);
+    EXPECT_TRUE(!img.empty());
+    CascadeClassifier face_cascade;
+    EXPECT_TRUE(face_cascade.load(filename));
+    EXPECT_NO_THROW(facemark = createFacemarkKazemi(face_cascade));
+    EXPECT_FALSE(facemark.empty());
+    EXPECT_NO_THROW(facemark->load(modelfilename));
```

Comment:
Remove `EXPECT_NO_THROW` wrapper (anywhere) to see the actual exception message.